src/metabase/query_processor/middleware/fetch_source_query.clj

```diff
@@ -1,40 +1,16 @@
 (ns metabase.query-processor.middleware.fetch-source-query
```

Comment:
Also rewritten from scratch, it was doing a lot of stuff that wasn't needed. New version using MLv2 tooling is ~150 LoC vs old version which was ~300 LoC.